src/mantine-core/src/components/Select/SelectDropdown/SelectDropdown.tsx

```diff
@@ -33,37 +37,78 @@ export const SelectDropdown = forwardRef<HTMLDivElement, SelectDropdownProps>(
       classNames,
       styles,
       dropdownComponent,
+      referenceElement,
+      direction = 'column',
+      onDirectionChange,
       __staticSelector,
     }: SelectDropdownProps,
     ref
   ) => {
     const { classes } = useStyles(null, { classNames, styles, name: __staticSelector });
 
     return (
-      <Transition
+      <Popper
+        referenceElement={referenceElement}
         mounted={mounted}
         transition={transition}
-        duration={transitionDuration}
-        timingFunction={transitionTimingFunction}
+        transitionDuration={transitionDuration}
+        transitionTimingFunction={transitionTimingFunction}
+        position="bottom"
+        modifiers={[
+          {
+            name: 'flip',
+            enabled: Boolean(onDirectionChange),
```

Comment:
should it be passed down to `Popper` component and boolean control prop `withFlip` or something like this?

Reply:
I think we should split this into two props:
- onDirectionChange
- allowDirectionChange (defaults to true)